openid-sharedsignals-eventdef-specification-1_0.md

```diff
@@ -0,0 +1,282 @@
+
+# Introduction {#introduction}
+
+Shared Signals Framework {{SSF}} enables sharing of signals and events between cooperating peers. It is currently leveraged by two applications – the Continuous Access Evaluation Protocol {{CAEP}} and Risk Incident Sharing and Coordination {{RISC}}.
+
+This specification defines how to translate normative SSF, CAEP, RISC event requirements into a JSON Schema. {{JSONSchema}} is a standardized way to describe the structure, constraints, and data types within a JSON document. JSON Schemas can also be used as {{JSONSchemaValidation}} to automatically check if a JSON document adheres to the defined schema, ensuring data integrity.
+
+Using JSON Schema to describe SSF has the following benefits:
+- Allows machine readability to auto generation of stubs.
+- It enables a faster process to create, update and get approval for new event types. 
+- JSON schema, rather than spec texts, is a more appropriate format to describe event types. 
+- It also allows event types to be versioned independently thus reducing the friction between the SSF core specification and event type publication.
+- Allows more events to be incorporated easily in the standards space beyond the usecases and charters of CAEP, RISC etc.
+
+# JSON Schema Defintion
+
+The following section describes how a map a SSF event to a JSON Schema Document.
+
+As defined in {{Section 4.3 of JSON Schema}}, a JSON Schema document, also called a "schema", is a JSON document used to describe another JSON Document, known as an instance.
+
+A JSON Schema document describes the instance of SSF SET event "payload" ({{Section 2 of SET}}). As such, the schema will defined the claims that pertian to the specific SSF event type. The $id for the schema document MUST be the same as the event identifier of the SET.
+
+
+The schema is made up of the following top-level JSON keys:
+
+\\$schema
+: REQUIRED, JSON string: URI of the version of JSON Schema that this document adheres to. E.g., `https://json-schema.org/draft/2020-12/schema`. Normative Schema Keyword.
+
+\\$id
+: REQUIRED, JSON string: Event Identifier of the SET and URI of the schema. MUST be publicly accessible or available out-of-band and resolve to JSON Schema document. Normative Schema Keyword.
+
+title
+: REQUIRED, JSON string: Title of the schema. Informational Schema Annotation.
+
+description
+: REQUIRED, JSON string: Intent of the schema. Informational Schema Annotation.
+
+type
+: REQUIRED JSON string: Defines the first constraint on the JSON data. MUST be "object". Normative Validation Keyword.
+
+properties
+: REQUIRED JSON object: Object that defines the keys in the JSON data that are being validated. Normative Validation Keyword.
+
+
+The following is a non-normative example of a JSON Schema document for a Session Revoked CAEP event.
+~~~ json
+{
+   "$schema":"https://json-schema.org/draft/2020-12/schema",
+   "$id":"https://schemas.openid.net/secevent/caep/event-type/session-revoked/v1.schema.json",
```

Comment:
I think we might need to define a mapping of the event identifier in the SET to the JSON schema identifier.

E.g. the event identifier is `$id` with `.schema.json` stripped?

And then we can say the inverse: an event's schema can be found by appending `.schema.json` to the event identifier and then dereferencing it.

Reply:
i think thats a good idea.